web/static/js/components/surveys/SurveyCard.jsx

```diff
@@ -77,106 +75,337 @@ class SurveyCard extends Component<any> {
+                <div className='grey-text card-description'>
+                  {survey.description}
+                </div>
+                <SurveyStatus survey={survey} short />
+              </Link>
+            </div>
+          </div>
+        </Card>
+      </div>
+    )
+  }
+}
 
-    const redirectTo = panelSurvey
-    ? routes.panelSurvey(panelSurvey.projectId, panelSurvey.id)
-    : routes.showOrEditSurvey(survey)
+const ActionMenu = (props) => {
+  const { actions } = props
 
-    const name = panelSurvey ? panelSurvey.name : survey.name
+  if (actions.length === 0) return null
 
-    const actionMenu = (
-      <Dropdown className='options' dataBelowOrigin={false} label={<i className='material-icons'>more_vert</i>}>
-        <DropdownItem className='dots'>
-          <i className='material-icons'>more_vert</i>
+  return (
+    <Dropdown className='options' dataBelowOrigin={false} label={<i className='material-icons'>more_vert</i>}>
+      <DropdownItem className='dots'>
+        <i className='material-icons'>more_vert</i>
+      </DropdownItem>
+      {actions.map((action, index) => (
+        <DropdownItem key={index}>
+          <a onClick={e => action.func()}>
+            <i className='material-icons'>folder</i>
+            {action.name}
+          </a>
         </DropdownItem>
-        {
-          this.movable()
-          ? <DropdownItem>
-            <a onClick={e => this.moveSurvey()}><i className='material-icons'>folder</i>{t('Move to')}</a>
-          </DropdownItem>
-          : null
-        }
-        {
-          this.deletable()
-            ? <DropdownItem>
-              <a onClick={e => this.deleteSurvey()}><i className='material-icons'>delete</i>{t('Delete')}</a>
-            </DropdownItem>
-            : null
-        }
-      </Dropdown>
-    )
+      ))}
+    </Dropdown>
+  )
+}
 
-    const surveyCard = <div className='survey-card'>
-      <Card>
-        <div className='card-content'>
-          <div className='survey-card-status'>
-            <Link className='grey-text' to={redirectTo}>
-              {t('{{percentage}}% of target completed', {percentage: String(Math.round(completionPercentage))})}
-            </Link>
-            { this.actionable() ? actionMenu : null }
-          </div>
-          <div className='card-chart'>
-            <RespondentsChart cumulativePercentages={cumulativePercentages} />
-          </div>
-          <div className='card-status'>
-            <Link className='card-title black-text truncate' title={name} to={redirectTo}>
-              <UntitledIfEmpty text={name} emptyText={untitledSurveyTitle(survey, t)} />
-            </Link>
-            <Link to={redirectTo}>
-              {description}
-              <SurveyStatus survey={survey} short />
-            </Link>
-          </div>
-        </div>
-      </Card>
-    </div>
+ActionMenu.propTypes = {
+  actions: PropTypes.array
+}
+
+class ConfirmationCheckModal extends Component<Props, State> {
```

Comment:
I'm not sure the naming is explicit on its differences with the regular ConfirmationModal component.

We're display a "warning" and need to tick a checkbox to "consent" with the action, so maybe something `WarningModal` or `ExplicitConfirmationModal` or something else?

Reply:
Both `ConfirmationModal` and `ConfirmationCheckModal` are warnings. The only difference is the extra step of clicking a check input to enable the confirmation button. So maybe we can call it `TwoStepsConfirmationModal`?

Reply:
Oh `TwoStepsConfirmationModal` is nice!